h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/matrix/SparseMatrixFactory.java

```diff
@@ -122,22 +115,22 @@ public static int initializeFromChunkIds(
 
         Frame _f;
         int[] _chunks;
-        Vec.Reader[] _vecs; 
-        Vec.Reader _w;
+        Vec[] _vecs; 
+        Vec _w;
         DataInfo _di;
         SparseMatrix _matrix;
         SparseMatrixDimensions _dims;
-        Vec.Reader _respVec;
+        Vec _respVec;
         float[] _resp; 
         float[] _weights;
         
         // OUT
         int[] _actualRows;
         
         InitializeCSRMatrixFromChunkIdsMrFun(
-            Frame f, int[] chunks, Vec.Reader[] vecs, Vec.Reader w, DataInfo di,
+            Frame f, int[] chunks, Vec[] vecs, Vec w, DataInfo di,
```

Comment:
Minor, let's not solve in this PR. I've noticed the vecs ALWAYS come from the frame. I guess handing over the `Frame f` is enough ?